cassandra-extension/deployment/src/main/java/com/datastax/oss/quarkus/deployment/CassandraClientProcessor.java

```diff
@@ -0,0 +1,181 @@
+ * limitations under the License.
+ */
+package com.datastax.oss.quarkus.deployment;
+
+import static io.quarkus.deployment.annotations.ExecutionTime.RUNTIME_INIT;
+import static io.quarkus.deployment.annotations.ExecutionTime.STATIC_INIT;
+
+import com.datastax.oss.driver.api.core.CqlSession;
+import com.datastax.oss.driver.internal.core.addresstranslation.PassThroughAddressTranslator;
+import com.datastax.oss.driver.internal.core.connection.ExponentialReconnectionPolicy;
+import com.datastax.oss.driver.internal.core.loadbalancing.DefaultLoadBalancingPolicy;
+import com.datastax.oss.driver.internal.core.metadata.MetadataManager;
+import com.datastax.oss.driver.internal.core.metadata.NoopNodeStateListener;
+import com.datastax.oss.driver.internal.core.metadata.schema.NoopSchemaChangeListener;
+import com.datastax.oss.driver.internal.core.retry.DefaultRetryPolicy;
+import com.datastax.oss.driver.internal.core.session.throttling.PassThroughRequestThrottler;
+import com.datastax.oss.driver.internal.core.specex.NoSpeculativeExecutionPolicy;
+import com.datastax.oss.driver.internal.core.time.AtomicTimestampGenerator;
+import com.datastax.oss.driver.internal.core.tracker.NoopRequestTracker;
+import com.datastax.oss.quarkus.config.CassandraClientConfig;
+import com.datastax.oss.quarkus.runtime.AbstractCassandraClientProducer;
+import com.datastax.oss.quarkus.runtime.CassandraClientRecorder;
+import io.quarkus.arc.Unremovable;
+import io.quarkus.arc.deployment.BeanContainerListenerBuildItem;
+import io.quarkus.arc.deployment.GeneratedBeanBuildItem;
+import io.quarkus.arc.deployment.GeneratedBeanGizmoAdaptor;
+import io.quarkus.deployment.annotations.BuildProducer;
+import io.quarkus.deployment.annotations.BuildStep;
+import io.quarkus.deployment.annotations.Record;
+import io.quarkus.deployment.builditem.ConfigurationBuildItem;
+import io.quarkus.deployment.builditem.FeatureBuildItem;
+import io.quarkus.deployment.builditem.nativeimage.NativeImageResourceBuildItem;
+import io.quarkus.deployment.builditem.nativeimage.ReflectiveClassBuildItem;
+import io.quarkus.deployment.builditem.nativeimage.RuntimeInitializedClassBuildItem;
+import io.quarkus.deployment.recording.RecorderContext;
+import io.quarkus.gizmo.ClassCreator;
+import io.quarkus.gizmo.ClassOutput;
+import io.quarkus.gizmo.MethodCreator;
+import io.quarkus.gizmo.MethodDescriptor;
+import io.quarkus.gizmo.ResultHandle;
+import io.quarkus.smallrye.health.deployment.spi.HealthBuildItem;
+import java.util.Arrays;
+import java.util.List;
+import javax.enterprise.context.ApplicationScoped;
+import javax.enterprise.inject.Default;
+import javax.enterprise.inject.Produces;
+
+class CassandraClientProcessor {
+  public static final String CASSANDRA_CLIENT = "cassandra-client";
+
+  @BuildStep
+  List<ReflectiveClassBuildItem> registerForReflection() {
+    return Arrays.asList(
+        new ReflectiveClassBuildItem(true, true, ExponentialReconnectionPolicy.class.getName()),
```

Comment:
If I understand correctly, this list contains the default implementations that the driver loads by reflection. But happens if the users configures a different class, for example `DcInferringLoadBalancingPolicy`?

Reply:
Yes, we need to add all of those, not only defaults - thanks for pointing that out.
I added all of those. One question regarding `LoadBalancingPolicies` - should we expose the two that are deprecated:
`DseDcInferringLoadBalancingPolicy` and `DseLoadBalancingPolicy`? I think we should not - we should discourage users to use it.

Reply:
I agree to not expose the deprecated ones.